common/ferrostar/src/models.rs

```diff
@@ -337,6 +333,15 @@ impl RouteStep {
             distance_to_end_of_step - instruction.trigger_distance_before_maneuver <= 5.0
         })
     }
+
+    /// Get the annotation data at a specific point along the step.
+    ///
+    /// `at_coordinate_index` is the index of the coordinate in the step geometry.
+    pub fn get_current_annotation_json(&self, at_coordinate_index: u64) -> Option<String> {
```

Comment:
Probably something along these lines is a bit clearer. Return type subject to pending other discussion.

```suggestion
    pub fn get_annotation_at_current_index(&self, at_coordinate_index: u64) -> Option<Vec<u8>> {
```